runtime/lua/vim/diagnostic.lua

```diff
@@ -1052,8 +1057,11 @@ function M._get_virt_text_chunks(line_diags, opts)
   -- TODO(tjdevries): Allow different servers to be shown first somehow?
   -- TODO(tjdevries): Display server name associated with these?
   if last.message then
+    if type(suffix) == 'function' then
+      suffix = suffix(last) or ''
+    end
     table.insert(virt_texts, {
-      string.format('%s %s', prefix, last.message:gsub('\r', ''):gsub('\n', '  ')),
+      string.format('%s %s%s', prefix, last.message:gsub('\r', ''):gsub('\n', '  '), suffix),
```

Comment:
`prefix` does not require a trailing space, so `suffix` should not require a leading space, for consistency (unless there is a compelling argument or use case otherwise?).

Reply:
Good point.  I guess one compelling reason is for consistency with `open_float()`'s suffix, which enables reusing the same `suffix` function for both virt_text and float.  But I don't feel strongly either way.

Reply:
That’s a valid reason, so let’s leave it as is. Ideally, `prefix` should not assume a trailing space either, but that ship has sailed.

Reply:
I think the reason `prefix` doesn't need a trailing space is because when a line has multiple diagnostics, it renders multiple prefixes (eg. `■■`)